crates/rpc-proto/src/lib.rs

```diff
@@ -0,0 +1,31 @@
+#![cfg_attr(not(feature = "std"), no_std)]
```

Comment:
Do we need to gate it by `std` feature? In other crates we just do `#![no_std]` and it seems to work.

Reply:
If you link `alloc` and use that, then it'd work. Here we can replace `String` and `format!` with their `alloc` counterparts, but not things like `fs`/`Path` so it would fail to compile correctly by default (since it would include the `std` feature).

Reply:
Ah - I see. Another way we should be able to do it is:
```rust
#![no_std]

#[cfg(feature = "std")]
extern crate std;
```
This is similar to how it is done in `miden-vm` [assembler](https://github.com/0xPolygonMiden/miden-vm/blob/main/assembly/src/lib.rs).

Reply:
This makes the function a bit more verbose with the imports since it seems the prelude is not loaded automatically, but it works fine.

Reply:
I think that's OK.